### PR TITLE
Add a method to force monoscopic rendering.

### PR DESCRIPTION
**Pull request by thomasflynn**
Add a method to force monoscopic rendering per request in Issue 75.

GearVRf-DCO-1.0-Signed-off-by: Thomas Flynn 
tom.flynn@samsung.com

### CODE REVIEW

File: GVRf/Framework/src/org/gearvrf/GVRActivity.java [hunk 1]
```diff
@@ -136,6 +137,24 @@ public void setScript(GVRScript gvrScript, String distortionDataFileName) {
         }
     }
 
+    /**
+     * Sets whether to force rendering to be single-eye, monoscopic view.
+     *
```
Comment (chenchao1407) [inline GVRf/Framework/src/org/gearvrf/GVRActivity.java]: I think we may need to either add JavaDoc here to specify this setForceMonoscopic only takes effect before you call setScript or throws an exception like UnsupportedOperationException if someone calls this after setScript().

**Comment by chenchao1407** [comment]
Codes looks almost fine and works as expected. Only a small concern on the JavaDoc comment.

**Comment by thomasflynn** [comment]
added the javadoc.

File: GVRf/Framework/src/org/gearvrf/GVRActivity.java [hunk 2]
```diff
@@ -136,6 +137,24 @@ public void setScript(GVRScript gvrScript, String distortionDataFileName) {
...
+    public void setForceMonoscopic(boolean force) {
+        forceMonoscopic = force;
+    }
+
+    /**
+     * Returns whether a monoscopic view will be created during {@linkplain setScript setScript()}.
+     */
+    public boolean getForceMonoscopic() {
+        return forceMonoscopic;
+    }
```
Comment (chenchao1407) [inline GVRf/Framework/src/org/gearvrf/GVRActivity.java]: Actually I think the logic here is also incorrect. What you should return is forceMonoscopic || !isVrSupported()

**Comment by thomasflynn** [comment]
clarified the javadoc in getForceMonoscopic.

**Comment by chenchao1407** [comment]
OK. I think it is good to me now.

**Comment by JonShemitz** [comment]
I’ll review it as soon as I can.

File: GVRf/Framework/src/org/gearvrf/GVRActivity.java [hunk 3]
```diff
@@ -45,6 +45,7 @@
 
     private GVRViewManager mGVRViewManager = null;
     private GVRCamera mCamera;
+    private boolean forceMonoscopic = false;
```
Comment (JonShemitz) [inline GVRf/Framework/src/org/gearvrf/GVRActivity.java]: Much as I dislike the `'m'` prefix, I think we should be internally consistent: when you add a new field to a block of fields, you probably should honor the local style.

File: GVRf/Framework/src/org/gearvrf/GVRActivity.java [hunk 4]
```diff
@@ -136,6 +137,25 @@ public void setScript(GVRScript gvrScript, String distortionDataFileName) {
         }
     }
 
+    /**
+     * Sets whether to force rendering to be single-eye, monoscopic view.
+     *
+     * @param force
+     *            If true, will create a GVRMonoscopicViewManager when {@linkplain setScript setScript()} is called.  If false, will proceed to auto-detect whether the device supports VR rendering and choose the appropriate ViewManager.  This call will only an effect if it is called before {@linkplain setScript setScript()}.
```
Comment (JonShemitz) [inline GVRf/Framework/src/org/gearvrf/GVRActivity.java]: {@link syntax is off, and text is not auto-formatted.

     *            If true, will create a GVRMonoscopicViewManager when
     *            {@linkplain #setScript(GVRScript, String) setScript()} is
     *            called. If false, will proceed to auto-detect whether the
     *            device supports VR rendering and choose the appropriate
     *            ViewManager. This call will only have an effect if it is
     *            called before {@linkplain #setScript(GVRScript, String)
     *            setScript()}.

File: GVRf/Framework/src/org/gearvrf/GVRActivity.java [hunk 5]
```diff
@@ -136,6 +137,25 @@ public void setScript(GVRScript gvrScript, String distortionDataFileName) {
...
+     *
+     */
+    public void setForceMonoscopic(boolean force) {
+        forceMonoscopic = force;
+    }
+
+    /**
+     * Returns whether a monoscopic view was asked to be forced during {@linkplain setScript setScript()}.
```
Comment (JonShemitz) [inline GVRf/Framework/src/org/gearvrf/GVRActivity.java]: Another invalid {@linkplain ...

**Comment by thomasflynn** [comment]
ok, renamed variable and fixed formatting and javadoc.

File: GVRf/Framework/src/org/gearvrf/GVRActivity.java [hunk 6]
```diff
@@ -136,6 +137,32 @@ public void setScript(GVRScript gvrScript, String distortionDataFileName) {
...
 
+    /**
+     * Sets whether to force rendering to be single-eye, monoscopic view.
+     * 
+     * @param force
+     *            If true, will create a GVRMonoscopicViewManager when
+     *            {@linkplain setScript setScript()} is called. If false, will
+     *            proceed to auto-detect whether the device supports VR
+     *            rendering and choose the appropriate ViewManager. This call
+     *            will only an effect if it is called before
```
Comment (JonShemitz) [inline GVRf/Framework/src/org/gearvrf/GVRActivity.java]: `will only an effect` s/b `will only have an effect`, but I think most people can deal with this sort of typo.

**Pull request merged**